crates/ide_completion/src/tests/item_list.rs

```diff
@@ -0,0 +1,156 @@
+use expect_test::expect;
+
+use crate::tests::check;
+
+#[test]
+fn in_mod_item_list() {
+    check(
+        r#"mod tests {
+    $0
+}
+"#,
+        expect![[r#"
+            kw pub(crate)
+            kw pub
+            kw unsafe
+            kw fn
+            kw const
+            kw type
+            kw impl
+            kw extern
+            kw use
+            kw trait
+            kw static
+            kw mod
+            kw enum
+            kw struct
+            kw union
+            sn tmod (Test module)
+            sn tfn (Test function)
+            sn macro_rules
+        "#]],
+    )
+}
+
+#[test]
+fn in_source_file_item_list() {
+    check(
+        r#"
+enum Enum { Variant }
+struct MyStruct {}
+#[macro_export]
+macro_rules! foo {}
+mod bar {}
+const CONST: () = ();
+
+$0"#,
+        expect![[r##"
+            kw pub(crate)
+            kw pub
+            kw unsafe
+            kw fn
+            kw const
+            kw type
+            kw impl
+            kw extern
+            kw use
+            kw trait
+            kw static
+            kw mod
+            kw enum
+            kw struct
+            kw union
+            sn tmod (Test module)
+            sn tfn (Test function)
+            sn macro_rules
+            md bar
+            ma foo!(…)          #[macro_export] macro_rules! foo
+            ma foo!(…)          #[macro_export] macro_rules! foo
```

Comment:
Interestingly enough we apparently complete `#[macro_export]` macros defined in the same scope like here twice, I assume the export puts it back into the scope a second time?

Reply:
This feels like a broken window maybe -- I recall setting up to fix duplicated macro completions at least twice. Both times I figured what the problem is, but I didn't fix it, as it was too annoying. Seems like it's faster to just fix it, rather than re-discover time and time again

Reply:
Ye I remember there being dupe macro completions, interestingly enough while this gives us duplicates the dupe doesn't show up in VSCode for me so I imagine VSCode is filtering dupes out? 
Will open an issue about this so its being tracked anyways 👍